modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase

```diff
@@ -560,6 +560,93 @@ definition {
 		}
 	}
 
+	@description = "LPS-178784 - Verify that it's possible to send an e-mail notification with Action (Download)"
+	@priority = 5
+	test CanSendEmailViaActionDownload {
+		property portal.acceptance = "true";
+
+		task ("Given an Email Notification Template") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Notifications",
+				panel = "Control Panel",
+				portlet = "Templates");
+
+			ObjectNotifications.addNewNotificationTemplate(
+				emailBody = "Email Body",
+				entryTo = "test@liferay.com",
+				fromAddress = "Liferay",
+				fromName = "Test Test",
+				name = "Notification Template",
+				notificationType = "Email",
+				subject = "Subject");
+
+			PortletEntry.save();
+		}
+
+		task ("And given a Custom Object that contains an attachment field with an action (On After Attachment Download)") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 178784",
+				objectName = "CustomObject178784",
+				pluralLabelName = "Custom Objects 178784");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Attachment",
+				fieldLabelName = "From Computer And Show Files In DM",
+				fieldName = "fromComputerAndShowFilesInDM",
+				fieldType = "Long",
+				fileSource = "userComputer",
+				isRequired = "false",
+				objectName = "CustomObject178784");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject178784");
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 178784");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.gotoAddObjectAction();
+
+			ObjectAdmin.fillObjectAction(
+				actionLabel = "Custom Action",
+				actionName = "customAction",
+				notificationName = "Notification Template",
+				thenAction = "Notification",
+				whenAction = "On After Attachment Download");
+
+			PortletEntry.save();
+		}
+
+		task ("And given an entry on the custom object") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject178784");
+
+			ObjectAutoGenerated.addAttachment(fileName = "LiferayLogo.png");
+
+			PortletEntry.save();
+		}
+
+		task ("When the entry is downloaded") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject178784");
+
+			ObjectPortlet.viewEntryDetails(entry = "LiferayLogo");
+
+			ObjectAutoGenerated.downloadFileOnAttachment(fileOnAttachment = "LiferayLogo");
+		}
+
+		task ("Then the email is sent immediately to Notification Queue") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Notifications",
+				panel = "Control Panel",
+				portlet = "Queue");
+
+			AssertElementPresent(
+				key_entryLabel = "CustomObject178784",
+				key_entryStatus = "Sent",
+				locator1 = "ObjectNotifications#VIEW_NOTIFICATION_ENTRIES");
```

Comment:
Can we change the steps so that we check the email using MockMock? We should have existing examples, @andreefarias can you help @victorhcunha if he needs help with this?